packages/flutter_availability_data_interface/lib/src/models/availability.dart

```diff
@@ -0,0 +1,92 @@
+/// A model defining the data structure for an availability
+class AvailabilityModel {
+  /// Creates a new availability
+  AvailabilityModel({
+    required this.userId,
+    required this.startDate,
+    required this.endDate,
+    required this.breaks,
+    this.id,
+  });
+
+  /// the identifier for this availability
+  final String? id;
+
+  /// The uniquely identifiable string for who or what the
+  final String userId;
+
+  /// The from date of this availability.
+  ///
+  /// [startDate] will always have to be before the end date.
+  final DateTime startDate;
+
+  /// The until date of this availability
+  ///
+  /// [endDate] will always be before the start date.
+  final DateTime endDate;
+
+  /// A list of breaks during the specified time period
+  final List<AvailabilityBreakModel> breaks;
+
+  /// Copies the current properties into a new instance of [AvailabilityModel],
+  /// except for the properties provided to this method.
+  AvailabilityModel copyWith({
+    String? id,
+    String? userId,
+    DateTime? startDate,
+    DateTime? endDate,
+    List<AvailabilityBreakModel>? breaks,
+  }) =>
+      AvailabilityModel(
+        id: id ?? this.id,
+        userId: userId ?? this.userId,
+        startDate: startDate ?? this.startDate,
+        endDate: endDate ?? this.endDate,
+        breaks: breaks ?? this.breaks,
+      );
+}
+
+/// A model defining the structure of a break within an [AvailabilityModel]
+class AvailabilityBreakModel {
+  /// Create a new AvailabilityBreakModel
+  AvailabilityBreakModel({
+    required this.startTime,
+    required this.endTime,
+    required this.duration,
```

Comment:
Any reason you're making it required even though it's nullable?

Reply:
I forgot some of the information on the model, will append it now